tt_metal/impl/device/device_pool.cpp

```diff
@@ -250,6 +250,28 @@ void DevicePool::add_devices_to_pool(std::vector<chip_id_t> device_ids) {
     }
 }
 
+void DevicePool::register_worker_thread_for_device(Device* device, uint64_t worker_thread_id) {
+    if (this->worker_thread_ids_per_device.find(device)  != this->worker_thread_ids_per_device.end()) {
+        TT_FATAL(worker_thread_ids_per_device.at(device) == worker_thread_id, "Cannot register more than one worker thread per device.");;
+    } else {
+        TT_FATAL(this->worker_thread_ids.find(worker_thread_id) == this->worker_thread_ids.end(), "Cannot register a single worker thread on multiple devices");
+    }
+
+    this->worker_thread_ids_per_device.insert({device, worker_thread_id});
+    this->worker_thread_ids.insert(worker_thread_id);
+}
+
+void DevicePool::delete_worker_thread_for_device(Device* device) {
+    if (this->worker_thread_ids_per_device.find(device) != this->worker_thread_ids_per_device.end()) {
+        this->worker_thread_ids.erase(this->worker_thread_ids_per_device.at(device));
```

Comment:
Ideally we should not be looking in the map twice, reusing the iterator returned by find.

Reply:
Done